plaidml/CMakeLists.txt

```diff
@@ -6,26 +6,61 @@ add_subdirectory(op)
 
 include(symbols.cmake)
 
-pml_cc_library(
-  NAME plaidml
-  TYPE SHARED
-  LINKOPTS
-    # TODO: use platform-specific flags
-    -Wl,--version-script ${LINUX_LDS}
-  PROPS
-    LINK_DEPENDS ${LINUX_LDS}
-    LINK_DEPENDS ${MACOS_LD}
+if (NOT WIN32)
+  pml_cc_library(
+    NAME plaidml
+    TYPE SHARED
+    LINKOPTS
+      # TODO: use platform-specific flags
+      -Wl,--version-script ${LINUX_LDS}
+    PROPS
+      LINK_DEPENDS ${LINUX_LDS}
+      LINK_DEPENDS ${MACOS_LD}
+      OUTPUT_NAME plaidml
+      SOVERSION 1
+    PRIVATE_DEPS
+      plaidml::core::obj
+      plaidml::edsl::obj
+      plaidml::exec::obj
+      plaidml::op::obj
+      pmlc::rt::all_runtimes
+      pmlc::target::all_obj
+  )
+else()
+# For Windows build both static and shared lib
+  pml_cc_library(
+    NAME plaidml
+    TYPE STATIC
+    LINKOPTS
+    PROPS
+      OUTPUT_NAME plaidml
+      SOVERSION 1
+    DEPS
+      plaidml::core::obj
+      plaidml::edsl::obj
+      plaidml::exec::obj
+      plaidml::op::obj
+      pmlc::rt::all_runtimes
+      pmlc::target::all_obj
+  )
+
+  pml_cc_library(
+    NAME plaidml_dll
+    TYPE SHARED
+    LINKOPTS
```

Comment:
I think we need to specify the `.def` file here, but I don't remember the exact flag syntax. You could just drop the `paidml_dll` target for now since we have nothing that tests it here anyways.